diff --git a/elks/Makefile b/elks/Makefile
index 3f7d69ea..ba9db29b 100644
--- a/elks/Makefile
+++ b/elks/Makefile
@@ -129,7 +129,7 @@ include/linuxmt/compiler-generated.h:
 	printf > include/linuxmt/compiler-generated.h		\
 		'#define UTS_RELEASE %s\n' "\"$(DIST)\""
 	printf >> include/linuxmt/compiler-generated.h		\
-		'#define UTS_VERSION %s\n' "\"hp200lx-3.5.2c 2026-09-28\""
+		'#define UTS_VERSION %s\n' "\"hp200lx-3.5.2d 2026-09-29\""
 
 #########################################################################
 # Specification files for archives.
diff --git a/elks/arch/i86/kernel/hp200lx-fw.c b/elks/arch/i86/kernel/hp200lx-fw.c
index 72f0b549..2c5e9e19 100644
--- a/elks/arch/i86/kernel/hp200lx-fw.c
+++ b/elks/arch/i86/kernel/hp200lx-fw.c
@@ -110,7 +110,7 @@ void hp200lx_fw_prepare(void)
     sys_caps &= ~(CAP_KBD_LEDS | CAP_IRQ2MAP9 | CAP_IRQ8TO15);
     disable_irq(1);
     disable_irq(2);
-    printk("HPR352C ROM %x IRQ2 %x:%x IER %x/%x PIC %x\n",
+    printk("HPR352D ROM %x IRQ2 %x:%x IER %x/%x PIC %x\n",
         hp_model_revision, hpfw_rom_vector[1], hpfw_rom_vector[0],
         hornet_read(0x18), hornet_read(0x19), inb(0x21));
 }
