--- a/elks/Makefile
+++ b/elks/Makefile
@@ -129,7 +129,7 @@
 	printf > include/linuxmt/compiler-generated.h		\
 		'#define UTS_RELEASE %s\n' "\"$(DIST)\""
 	printf >> include/linuxmt/compiler-generated.h		\
-		'#define UTS_VERSION %s\n' "\"hp200lx-r3.5 2026-09-28\""
+		'#define UTS_VERSION %s\n' "\"hp200lx-3.5.1-b1 2026-09-28\""
 
 #########################################################################
 # Specification files for archives.
--- a/elks/arch/i86/kernel/hp200lx-fw.c
+++ b/elks/arch/i86/kernel/hp200lx-fw.c
@@ -76,7 +76,7 @@
     sys_caps &= ~(CAP_KBD_LEDS | CAP_IRQ2MAP9 | CAP_IRQ8TO15);
     disable_irq(1);
     disable_irq(2);
-    printk("HPR3.5 ROM %x IRQ2 %x:%x IER %x/%x PIC %x\n",
+    printk("HPR351B1 ROM %x IRQ2 %x:%x IER %x/%x PIC %x\n",
         hp_model_revision, hpfw_rom_vector[1], hpfw_rom_vector[0],
         hornet_read(0x18), hornet_read(0x19), inb(0x21));
 }
